.changeset/smart-bobcats-obey.md

```
@@ -0,0 +1,7 @@
+---
+'@mastra/core': patch
+'@mastra/libsql': patch
+'@mastra/pg': patch
+---
+
+Fixed listing stored agents, skills, prompt blocks, scorers, MCP clients/servers and workspaces issuing one version query per entity. `listResolved()` now fetches all active versions in a single batch through a new overridable `getVersions(ids)` method on versioned storage domains, so list endpoints stay fast on database-backed storage. Fixes https://github.com/mastra-ai/mastra/issues/22524
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)